src/main/scala/tutorial/Serve.scala

```diff
@@ -50,36 +50,47 @@ class NDVIServiceActor extends Actor with HttpService {
     ColorMap.fromStringDouble(ConfigFactory.load().getString("tutorial.colormap")).get
 
   def root =
-    pathPrefix(IntNumber / IntNumber / IntNumber) { (zoom, x, y) =>
-      respondWithMediaType(MediaTypes.`image/png`) {
-        complete {
-          future {
+    //pathPrefix(nd) {
 
-            // Read in the tile at the given z/x/y coordinates.
-            val tileOpt: Option[MultibandTile] =
+
+      pathPrefix(Segment / IntNumber / IntNumber / IntNumber) { (nd, zoom, x, y) =>
+        respondWithMediaType(MediaTypes.`image/png`) {
+          complete {
+            future {
+
+              // Read in the tile at the given z/x/y coordinates.
+              val tileOpt: Option[MultibandTile] =
               try {
-                Some(ServeNDVI.reader(LayerId("landsat",zoom)).read(x, y))
+                Some(Serve.reader(LayerId("landsat", zoom)).read(x, y))
               } catch {
                 case _: TileNotFoundError =>
                   None
               }
-
-            tileOpt.map { tile =>
-              // Compute the NDVI
-              val ndvi =
-                tile.convert(DoubleConstantNoDataCellType).combineDouble(0, 1) { (r, ir) =>
-                  if(isData(r) && isData(ir)) {
-                    (ir - r) / (ir + r)
-                  } else {
-                    Double.NaN
-                  }
+              if (nd == "ndvi") {
```

Comment:
It would be clearer to use `match` here so we can match on `ndvi` and `ndwi` explicitly. Then we can have default case as well with `case _ =>`